manifests/system.pp

```diff
@@ -67,11 +56,13 @@
 
   }
   else {
+    ensure_packages(['curl'])
```

Comment:
> Also there is no point in checking whether the package curl defined before doing an ensure packages.

@ColinHebert What about non linux operating systems?  Presumably the `default: { }` exists for a reason?

Would adding an `else` to the existing `if !defined` solve your problem? eg
```
if ! defined(Package['curl']) {
  case $::kernel {
    'Linux': {
       ensure_packages(['curl'])
       Package['curl'] -> Exec['system-rvm']	
    }
    default: { }
} else {
  Package['curl'] -> Exec['system-rvm']
}

Reply:
I've opened https://github.com/voxpupuli/puppet-rvm/pull/181 as a replacement. It doesn't address the concerns you mentioned here, but I think it's safe enough.